numpy/core/shape_base.py

```diff
@@ -282,6 +282,10 @@ def hstack(tup):
 
     """
     arrs = [atleast_1d(_m) for _m in tup]
+
+    if len(arrs) == 0:
+        raise ValueError('need at least one element to concatenate')
+
     # As a special case, dimension 0 of 1-dimensional arrays is "horizontal"
     if arrs[0].ndim == 1:
```

Comment:
An alternative fix would be to simply change this line to `arrs and arrs[0].ndim == 1`